### PR TITLE
Feature/17 import geonames

### PR DESCRIPTION
**Pull request by sastafford**
@rjrudin - sorry first pull request compared to master, this one compares to dev

### CODE REVIEW

File: examples/geonames/src/main/java/com/marklogic/example/IngestGeonamesToMarkLogicConfig.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    @Bean
+    @JobScope
+    protected Step step1(ItemReader<Geoname> reader, ItemProcessor<Geoname, Document> processor,
+    ItemWriter<Document> writer, @Value("#{jobParameters['chunk'] ?: 100}") Integer chunkSize) {
+        return stepBuilderFactory.get("step1")
+                .<Geoname, Document> chunk(chunkSize)
+                .reader(reader)
+                .processor(processor)
+                .writer(writer)
+                .taskExecutor(taskExecutor())
```
Comment (sastafford) [inline examples/geonames/src/main/java/com/marklogic/example/IngestGeonamesToMarkLogicConfig.java]: Experimenting with a taskExecutor to try and parallelize processing.  My initial chunk size was 10 with no taskExecutor, total time was > 1 minute.  With a chunk size of 100 and this taskExecutor was able to get around 35 seconds.

File: examples/geonames/src/main/java/com/marklogic/example/geonames/GeonamesItemProcessor.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.springframework.batch.item.ItemProcessor;
+import org.w3c.dom.Document;
+
+public class GeonamesItemProcessor implements ItemProcessor<Geoname, Document> {
+
+	protected JAXBContext jaxbContext() {
+		JAXBContext jaxbContext;
+		try {
+			jaxbContext = JAXBContext.newInstance(Geoname.class);
```
Comment (sastafford) [inline examples/geonames/src/main/java/com/marklogic/example/geonames/GeonamesItemProcessor.java]: This is an example of POJO to org.w3c.Document.

File: examples/geonames/src/main/java/com/marklogic/example/geonames/GeonamesItemProcessor.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		Document doc = documentBuilder().newDocument();
+		try {
+			Marshaller marshaller = jaxbContext().createMarshaller();
+			marshaller.marshal(item, doc);
+		} catch (JAXBException e) {
+			e.printStackTrace();
+		}
+		
+		//Set document URI
+		doc.setDocumentURI("http://geonames.org/geoname/" + item.getId());
```
Comment (sastafford) [inline examples/geonames/src/main/java/com/marklogic/example/geonames/GeonamesItemProcessor.java]: I like the fact that I can set the URI on a org.w3c.Document.  Would be useful on a JSON generic object

File: examples/geonames/src/test/resources/logback.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<configuration>
```
Comment (sastafford) [inline examples/geonames/src/test/resources/logback.xml]: I noticed the default logging was DEBUG.  Default should be INFO

File: examples/geonames/gradle.properties [hunk 5]
```diff
@@ -0,0 +1,10 @@
+mlHost=localhost
+mlAppName=marklogic-spring-batch-test
+mlJobRepositoryName=marklogic-spring-batch-test
+mlRestPort=8200
+mlUsername=admin
+mlPassword=admin
+mlManageUsername=admin
```
Comment (rjrudin) [inline examples/geonames/gradle.properties]: You don't need the manage username/password if it's the same as mlUsername/mlPassword

File: examples/geonames/src/main/java/com/marklogic/example/IngestGeonamesToMarkLogicConfig.java (hunk 1 above)
Comment (rjrudin) [inline examples/geonames/src/main/java/com/marklogic/example/IngestGeonamesToMarkLogicConfig.java]: Interesting - is it reading from a single file or a directory of files? If it's a single file, I guess SB has support for handing each line off to a separate thread.

**Pull request merged**